workflow/controller/controller.go

```diff
@@ -417,6 +421,17 @@ func (wfc *WorkflowController) processNextItem() bool {
 	return true
 }
 
+func (wfc *WorkflowController) addingWorkflowDefaultValueIfValueNotExist(wf *wfv1.Workflow) {
+	//var workflowSpec *wfv1.WorkflowSpec = &wf.Spec
+	if wfc.Config.WorkflowDefaults != nil {
+		if err := mergo.Merge(&wf.Spec, *wfc.Config.WorkflowDefaults); err != nil {
```

Comment:
I think this there should probably be a library within Kubernetes than performs a strategic-merge-patch. I think you need to convert them both to JSON, apply the merge. This should probably be very similar to how Kustomize does it.


https://github.com/kubernetes/apimachinery/blob/master/pkg/util/strategicpatch/patch.go
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/strategicpatch/patch.go

Reply:
Great, I was very unsure about how to do this in the best way!